scipy/linalg/basic.py

```diff
@@ -34,10 +40,13 @@ def _solve_check(n, info, lamch=None, rcond=None):
         return
     E = lamch('E')
     if rcond < E:
+        orig_formatwarning = warnings.formatwarning
+        warnings.formatwarning = _truncated_warning_msg
```

Comment:
This stuff is not ok (and not really required afaics after setting the stacklevel correctly).

Reply:
I've picked it up from the official docs as the recommended practice to keep things local

Reply:
can you show where it's recommended?

in any case, I don't think it's required here, and moreover even if it is, it's not scipy's job to prettify the warning display

Reply:
Here it is https://docs.python.org/3.6/library/warnings.html#warnings.formatwarning

But I get your point then we can close this PR.